src/tribler-core/tribler_core/modules/metadata_store/community/gigachannel_community.py

```diff
@@ -131,13 +131,18 @@ def notify_gui(_, processing_results):
 
 class GigaChannelCommunity(NonLegacyGigaChannelCommunity):
     def __init__(self, *args, **kwargs):
+        self.components = kwargs.pop('components', None)
         super().__init__(*args, **kwargs)
 
         # Register legacy payload
         self.add_message_handler(LegacySelectResponsePayload, self.legacy_on_remote_select_response)
-
         self.new_style_peers = set()
 
+        # apply components
+        if self.components:
```

Comment:
It is better to move it into `NonLegacyGigaChannelCommunity`. Otherwise, we may accidentally delete it when removing the legacy stuff.

Reply:
Yes, sure, thank you :)